service/account/internal/src/main/java/org/eclipse/kapua/service/account/internal/AccountServiceImpl.java

```diff
@@ -169,13 +169,33 @@ public Account update(Account account) throws KapuaException {
             authorizationService.checkPermission(permissionFactory.newPermission(AccountDomains.ACCOUNT_DOMAIN, Actions.write, account.getScopeId()));
         }
 
+        if (account.getExpirationDate() != null) {
+            SystemSetting setting = SystemSetting.getInstance();
+            //check if the updated account is an admin account
+            if (setting.getString(SystemSettingKey.SYS_ADMIN_ACCOUNT).equals(account.getName())) {
+                //throw exception if trying to set an expiration date for an admin account
+                throw new KapuaIllegalArgumentException("notAllowedExpirationDate", account.getExpirationDate().toString());
```

Comment:
Please change the `throw` of this exception.
KapuaIllegalArgumentException should be used to mark invalid fields by their name. 
So either do

- throw new KapuaIllegalArgumentException("account.expirationDate", account.getExpirationDate().toString());

or

- throw new KapuaAccountException(KapuaAccountErrorCodes.OPERATION_NOT_ALLOWED, null, "Admin account cannot have an expiration date set");

The latter gives more information to the user on what's wrong instead of just only saying that the value of `account.expirationDate` is invalid.